### PR TITLE
SRCH-3830: using json for .eslintrc rules

### PR DESCRIPTION
**Pull request by yogesh27**
## Summary
- Updated .eslintrc from yaml to json
- Added/Modified .eslintrc rules for typescript 
- Update codeclimate.yml file to run eslint for typescript files
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,6 +58,15 @@
     "@pmmmwh/react-refresh-webpack-plugin": "^0.5.8",
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
+    "@typescript-eslint/eslint-plugin": "^5.0.0",
+    "@typescript-eslint/parser": "^5.54.1",
+    "eslint": "^8.0.1",
```
Comment (ddzz) [inline package.json:63]: Are these dependencies needed if we're not running ESLint through yarn?

Comment (yogesh27) [reply package.json:63]: I don't think so. These dev dependencies should be needed if we linting the code locally.

Comment (ddzz) [reply package.json:63]: I think these should be taken out. They are not needed if you lint locally with the CodeClimate CLI.

**Review by ddzz — APPROVED** [review]
LGTM

**Pull request merged**